### ISSUE TITLE
extra newlines in filterstream results

### ISSUE THREAD

**Issue by edsu** [gwu-libraries/social-feed-manager#321]
It looks like extra newlines are being added to the fillterstream results, e.g.:

```
{"created_at":"Thu Mar 12 17:57:05 +0000 2015","id":576079477735755776,"id_str":"576079477735755776","text":"RT @abo_hasan_111: \u0623\u0645\u0631\u064a\u0643\u0627 \u0627\u0644\u062a\u064a \u062a\u0641\u0627\u0647\u0645\u062a \u0645\u0639 \u0625\u064a\u0631\u0627\u0646 \u0644\u0627\u062d\u062a\u0644\u0627\u0644 \u0627\u0644\u0639\u0631\u0627\u0642 \u0648\u062d\u0645\u0627\u064a\u0629 \u0627\u0644\u0645\u0627\u0644\u0643\u064a \u0644\u0646 \u062a\u0642\u0641 \u0636\u062f \u0628\u0634\u0627\u0631 \u0648\u0627\u0644\u0623\u0646\u0638\u0645\u0629 \u0627\u0644\u0639\u0631\u0628\u064a\u0629 \u0627\u0644\u062a\u064a \u0633\u0644\u0645\u062a \u0627\u0644\u0639\u0631\u0627\u0642 \u0644\u0644\u0631\u0627\u0641\u0636\u0629 \u062a\u0633\u0644\u0645 \u2026","source":"\u003ca href=\"http:\/\/twitter.com\/download\/android\" rel=\"nofollow\"\u003eTwitter for Android\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":795526735,"id_str":"795526735","name":"\u0627\u0644\u0641\u0642\u064a\u0640\u0640\u0631 \u0625\u0644\u0649 \u0627\u0644\u0644\u0640\u0647","screen_name":"alfaqer87","location":"\u0646\u062d\u0640\u0640\u0646\u064f \u0644\u0634\u0631\u064a\u0639\u0640\u0629 \u0641\u0640\u062f\u0627\u0621","url":"http:\/\/www.jalnosra.com","description":"\u0627\u0633\u062a\u0648\u062f\u0639 \u0627\u0644\u0644\u0647 \u0635\u062d\u0628\u0627\u064b \u0643\u0646\u062a \u0627\u0630\u062e\u0631\u0647\u0645 .. \u0644\u0646\u0623\u0628\u0627\u062a \u0644\u0646\u0627 \u0625\u0646\u0633 \u0648\u0633\u0645\u0627\u0631 \n\u0627\u0644\u0645\u0644\u062a\u0642\u0649 \u0641\u064a \u062c\u0646\u0629 \u0627\u0644\u062e\u0644\u062f \u0627\u0646 \u0642\u0628\u0644\u062a ... \u0645\u0646\u0627 \u0635\u0644\u0627\u0629 \u0648\u0637\u0627\u0639\u0627\u062a \u0648\u0623\u0630\u0643\u0627\u0631","protected":false,"verified":false,"followers_count":2619,"friends_count":175,"listed_count":7,"favourites_count":286,"statuses_count":26542,"created_at":"Sat Sep 01 07:07:05 +0000 2012","utc_offset":10800,"time_zone":"Riyadh","geo_enabled":false,"lang":"ar","contributors_enabled":false,"is_translator":false,"profile_background_color":"ACDED6","profile_background_image_url":"http:\/\/pbs.twimg.com\/profile_background_images\/378800000053079454\/be313f25bc7cede44fbc831d174a8be5.png","profile_background_image_url_https":"https:\/\/pbs.twimg.com\/profile_background_images\/378800000053079454\/be313f25bc7cede44fbc831d174a8be5.png","profile_background_tile":false,"profile_link_color":"038543","profile_sidebar_border_color":"FFFFFF","profile_sidebar_fill_color":"F6F6F6","profile_text_color":"333333","profile_use_background_image":true,"profile_image_url":"http:\/\/pbs.twimg.com\/profile_images\/548565424212676608\/6ntEMIqz_normal.jpeg","profile_image_url_https":"https:\/\/pbs.twimg.com\/profile_images\/548565424212676608\/6ntEMIqz_normal.jpeg","profile_banner_url":"https:\/\/pbs.twimg.com\/profile_banners\/795526735\/1406288305","default_profile":false,"default_profile_image":false,"following":null,"follow_request_sent":null,"notifications":null},"geo":null,"coordinates":null,"place":null,"contributors":null,"retweeted_status":{"created_at":"Thu Mar 12 13:48:25 +0000 2015","id":576016899286302721,"id_str":"576016899286302721","text":"\u0623\u0645\u0631\u064a\u0643\u0627 \u0627\u0644\u062a\u064a \u062a\u0641\u0627\u0647\u0645\u062a \u0645\u0639 \u0625\u064a\u0631\u0627\u0646 \u0644\u0627\u062d\u062a\u0644\u0627\u0644 \u0627\u0644\u0639\u0631\u0627\u0642 \u0648\u062d\u0645\u0627\u064a\u0629 \u0627\u0644\u0645\u0627\u0644\u0643\u064a \u0644\u0646 \u062a\u0642\u0641 \u0636\u062f \u0628\u0634\u0627\u0631 \u0648\u0627\u0644\u0623\u0646\u0638\u0645\u0629 \u0627\u0644\u0639\u0631\u0628\u064a\u0629 \u0627\u0644\u062a\u064a \u0633\u0644\u0645\u062a \u0627\u0644\u0639\u0631\u0627\u0642 \u0644\u0644\u0631\u0627\u0641\u0636\u0629 \u062a\u0633\u0644\u0645 \u0623\u064a\u0636\u0627\u064b \u0633\u0648\u0631\u064a\u0627 .","source":"\u003ca href=\"http:\/\/twitter.com\/download\/iphone\" rel=\"nofollow\"\u003eTwitter for iPhone\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":3062784830,"id_str":"3062784830","name":"\u0623\u0628\u0648 \u062d\u0633\u0646 \u0627\u0644\u0643\u0648\u064a\u062a\u064a","screen_name":"abo_hasan_111","location":"","url":null,"description":"\u062d\u064a\u0646\u0645\u0627 \u0623\u0639\u0644\u0645 \u0623\u0646\u064a \u0639\u0644\u0649 \u0627\u0644\u062d\u0642 \u0641\u0644\u0627 \u0623\u0628\u0627\u0644\u064a \u0645\u0646 \u0643\u0627\u0646 \u062e\u0635\u0645\u0649 \n\u0639\u0644\u064a \u0628\u0646 \u062d\u0645\u062f \u0628\u0646 \u062d\u0633\u0646 \u0627\u0644\u0639\u0631\u062c\u0627\u0646\u064a","protected":false,"verified":false,"followers_count":2525,"friends_count":64,"listed_count":13,"favourites_count":4,"statuses_count":183,"created_at":"Thu Mar 05 11:39:36 +0000 2015","utc_offset":null,"time_zone":null,"geo_enabled":false,"lang":"ar","contributors_enabled":false,"is_translator":false,"profile_background_color":"C0DEED","profile_background_image_url":"http:\/\/abs.twimg.com\/images\/themes\/theme1\/bg.png","profile_background_image_url_https":"https:\/\/abs.twimg.com\/images\/themes\/theme1\/bg.png","profile_background_tile":false,"profile_link_color":"0084B4","profile_sidebar_border_color":"C0DEED","profile_sidebar_fill_color":"DDEEF6","profile_text_color":"333333","profile_use_background_image":true,"profile_image_url":"http:\/\/pbs.twimg.com\/profile_images\/573452466596929536\/NEiuWmlU_normal.jpeg","profile_image_url_https":"https:\/\/pbs.twimg.com\/profile_images\/573452466596929536\/NEiuWmlU_normal.jpeg","profile_banner_url":"https:\/\/pbs.twimg.com\/profile_banners\/3062784830\/1425556689","default_profile":true,"default_profile_image":false,"following":null,"follow_request_sent":null,"notifications":null},"geo":null,"coordinates":null,"place":null,"contributors":null,"retweet_count":5,"favorite_count":8,"entities":{"hashtags":[],"trends":[],"urls":[],"user_mentions":[],"symbols":[]},"favorited":false,"retweeted":false,"possibly_sensitive":false,"filter_level":"low","lang":"ar"},"retweet_count":0,"favorite_count":0,"entities":{"hashtags":[],"trends":[],"urls":[],"user_mentions":[{"screen_name":"abo_hasan_111","name":"\u0623\u0628\u0648 \u062d\u0633\u0646 \u0627\u0644\u0643\u0648\u064a\u062a\u064a","id":3062784830,"id_str":"3062784830","indices":[3,17]}],"symbols":[]},"favorited":false,"retweeted":false,"possibly_sensitive":false,"filter_level":"low","lang":"ar","timestamp_ms":"1426183025188"}

{"created_at":"Thu Mar 12 17:57:05 +0000 2015","id":576079478167732224,"id_str":"576079478167732224","text":"RT @mmfa: The parallels between the GOP's Iran letter and Pelosi's 2007 Syria visit are \"tenuous, at best.\" Here's why: http:\/\/t.co\/EMvtjcp\u2026","source":"\u003ca href=\"http:\/\/twitter.com\/download\/iphone\" rel=\"nofollow\"\u003eTwitter for iPhone\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":20433598,"id_str":"20433598","name":"DE Mom","screen_name":"Momof2inDE","location":"Coastal Delaware","url":null,"description":"Proud wife, mom, runner and political junkie living in coastal Delaware.","protected":false,"verified":false,"followers_count":947,"friends_count":1619,"listed_count":26,"favourites_count":1433,"statuses_count":10584,"created_at":"Mon Feb 09 13:09:53 +0000 2009","utc_offset":-14400,"time_zone":"Eastern Time (US & Canada)","geo_enabled":false,"lang":"en","contributors_enabled":false,"is_translator":false,"profile_background_color":"FF6699","profile_background_image_url":"http:\/\/abs.twimg.com\/images\/themes\/theme11\/bg.gif","profile_background_image_url_https":"https:\/\/abs.twimg.com\/images\/themes\/theme11\/bg.gif","profile_background_tile":true,"profile_link_color":"B40B43","profile_sidebar_border_color":"CC3366","profile_sidebar_fill_color":"E5507E","profile_text_color":"362720","profile_use_background_image":true,"profile_image_url":"http:\/\/pbs.twimg.com\/profile_images\/3338396379\/4a935281a1bd22be68bc8b4ab9f2c4ce_normal.jpeg","profile_image_url_https":"https:\/\/pbs.twimg.com\/profile_images\/3338396379\/4a935281a1bd22be68bc8b4ab9f2c4ce_normal.jpeg","default_profile":false,"default_profile_image":false,"following":null,"follow_request_sent":null,"notifications":null},"geo":null,"coordinates":null,"place":null,"contributors":null,"retweeted_status":{"created_at":"Thu Mar 12 17:50:02 +0000 2015","id":576077706841948160,"id_str":"576077706841948160","text":"The parallels between the GOP's Iran letter and Pelosi's 2007 Syria visit are \"tenuous, at best.\" Here's why: http:\/\/t.co\/EMvtjcpJUH","source":"\u003ca href=\"https:\/\/about.twitter.com\/products\/tweetdeck\" rel=\"nofollow\"\u003eTweetDeck\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":13493302,"id_str":"13493302","name":"Media Matters","screen_name":"mmfa","location":"Washington, DC","url":"http:\/\/mediamatters.org\/","description":"Media Matters for America is the nation's premier progressive media watchdog, research and information center.","protected":false,"verified":true,"followers_count":175204,"friends_count":2581,"listed_count":7611,"favourites_count":1230,"statuses_count":37060,"created_at":"Thu Feb 14 22:48:21 +0000 2008","utc_offset":-14400,"time_zone":"Eastern Time (US & Canada)","geo_enabled":false,"lang":"en","contributors_enabled":false,"is_translator":false,"profile_background_color":"F7F2F7","profile_background_image_url":"http:\/\/pbs.twimg.com\/profile_background_images\/660309504\/jywz9f77bmscs5ckmme9.png","profile_background_image_url_https":"https:\/\/pbs.twimg.com\/profile_background_images\/660309504\/jywz9f77bmscs5ckmme9.png","profile_background_tile":true,"profile_link_color":"0084B4","profile_sidebar_border_color":"FFFFFF","profile_sidebar_fill_color":"DDEEF6","profile_text_color":"333333","profile_use_background_image":true,"profile_image_url":"http:\/\/pbs.twimg.com\/profile_images\/420922743345782784\/Bjk2qDue_normal.jpeg","profile_image_url_https":"https:\/\/pbs.twimg.com\/profile_images\/420922743345782784\/Bjk2qDue_normal.jpeg","profile_banner_url":"https:\/\/pbs.twimg.com\/profile_banners\/13493302\/1399486776","default_profile":false,"default_profile_image":false,"following":null,"follow_request_sent":null,"notifications":null},"geo":null,"coordinates":null,"place":null,"contributors":null,"retweet_count":5,"favorite_count":2,"entities":{"hashtags":[],"trends":[],"urls":[{"url":"http:\/\/t.co\/EMvtjcpJUH","expanded_url":"http:\/\/mm4a.org\/1ED6XgA","display_url":"mm4a.org\/1ED6XgA","indices":[110,132]}],"user_mentions":[],"symbols":[]},"favorited":false,"retweeted":false,"possibly_sensitive":false,"filter_level":"low","lang":"en"},"retweet_count":0,"favorite_count":0,"entities":{"hashtags":[],"trends":[],"urls":[{"url":"http:\/\/t.co\/EMvtjcpJUH","expanded_url":"http:\/\/mm4a.org\/1ED6XgA","display_url":"mm4a.org\/1ED6XgA","indices":[120,140]}],"user_mentions":[{"screen_name":"mmfa","name":"Media Matters","id":13493302,"id_str":"13493302","indices":[3,8]}],"symbols":[]},"favorited":false,"retweeted":false,"possibly_sensitive":false,"filter_level":"low","lang":"en","timestamp_ms":"1426183025291"}
```

This makes counting the number of tweets in a file, and processing the line oriented json a bit more tedious than it needs to be.

**Comment by kerchner** [comment]
@edsu I'm unable to reproduce the problem.  Are you seeing it in every tweet written to the zip files by ```filterstream```?

**Comment by kerchner** [comment]
@edsu are you just referring to the newline in between each tweet? (rather than some sort of newline that doesn't belong in the midst of a single tweet)    If it's a newline between each tweet, I believe those were initially placed there for readability, but we can reconsider.

**Comment by dchud** [comment]
I don't recall why I added those.  For streams with a large volume of tweets, the API inserts (or used to at least) lines between tweets consisting of an integer indicating the number of tweets that were skipped while limiting the stream to a certain subset of all matching tweets.  This only applied to large volume tweets, so maybe I added the blank lines to have some consistency.  Or maybe it was just a mistake.

Either way I'm confident it wasn't for readability, because who reads these things? :)
